### DISCUSSION TITLE
Install LLVM's libclang library

### DISCUSSION THREAD

**Discussion by giuseppe998e** [spack/spack#36558] [Q&A] [answered]
Hello,

I was wondering if it's possible to install only the `libclang` library  from the `LLVM` package using Spack.  
Currently, I'm installing the entire package which is unnecessary for my specific use case with the Rust bindgen crate.

Thank you in advance.

**Comment by psakievich** [comment, marked as answer]
It is not currently available. There have been other requests like this in the past to install subsets of the LLVM package, but as far as I know there hasn't been any movement on that yet.  LLVM is a single entity in spack at this point. I think to break it up we would need to start creating smaller packages under the LLVM umbrella.

**Reply by giuseppe998e** [reply]
Is it not possible to do this with flags?  
I'm not saying now, indent in the future, instead of creating separate packages.

**Reply by psakievich** [reply]
Sure you could create variants for the subset features of LLVM too. I guess it just depends on how many flavors people want, the number of use case, and the build procedure. I am not an LLVM expert by any means. I usually just build it to get a clang compiler and that is all.
